src/QMCDrivers/tests/test_DescentEngine.cpp

```diff
@@ -30,6 +30,7 @@ TEST_CASE("DescentEngine RMSprop update","[drivers][descent]")
 
   Communicate myComm;
```

Comment:
Change to `Communicate* c = OHMMS::Controller;` and pass c to the constructor. This may solve your MPI issue in unit test.

Reply:
Thanks Ye. This fixes the MPI issue in the unit test so I've removed the check in DescentEngine.